node/silkworm/stagedsync/stage_execution.hpp

```diff
@@ -41,11 +42,14 @@ class Execution final : public IStage {
   private:
     std::unique_ptr<consensus::IEngine> consensus_engine_;
     BlockNum block_num_{0};
+    std::vector<Block> prefetched_blocks_;
```

Comment:
Because of the size of `Block` I wonder if it should not be `std::vector<unique_ptr<Block>>`

Reply:
The idea is to reuse the same vector of prefetched blocks in the loop in `Execution::execute_batch` so that `db::read_block` uses an already allocated `Block` instance. I'm not sure how an extra level of memory indirection (`std::vector<unique_ptr<Block>>`) could help.

Reply:
I was writing this because the prefetch_blocks function is called with `from` and `to` parameters and resizes the vector. I thought that maybe the resize could become slow if the vector becomes very large and prefect_blocks is called often with a small `to` - `from`, but maybe it is not an issue.

Reply:
> I was writing this because the prefetch_blocks function is called with `from` and `to` parameters and resizes the vector. I thought that maybe the resize could become slow if the vector becomes very large and prefect_blocks is called often with a small `to` - `from`, but maybe it is not an issue.

In reality the resizing happens only 2 times (when large segments of blocks gets executed) or even 1 time (likely when at the tip of the chain) cause `prefetch` fills a batch of blocks up to segment width or up to max_blocks (whichever the smaller)

Reply:
> In reality the resizing happens only 2 times

Oh great then my comment is irrelevant!